src/common.hpp

```diff
@@ -9,22 +9,84 @@
 #include <cinttypes>
 #include <cstddef>
 #include <cstdint>
+#include <cstdlib>
 #include <cstring>
 
 #include <array>
+#include <memory>
 #include <string>
 #include <unordered_map>
 #include <utility>
 #include <vector>
 
+/***************************** Helper types **********************************/
```

Comment:
(for me) why do we need this?

Reply:
Look at the change in processing.cpp. It’s like passing Iterable in Python.